tests/units/classes/test.php

```diff
@@ -466,6 +479,22 @@ public function testGetTestedClassPath()
 			;
 		}
 
+		public function testCallStaticOnTestedClass()
+		{
+			$this
+				->if($this->setTestedClassName('\mageekguy\atoum\withStatic'))
+				->then
+					->string($this->callStaticOnTestedClass('staticMethod', $return = uniqid()))->isEqualTo($return)
```

Comment:
The `atoum\test` class is one of the most complicated class to test. Even if your feature looks quite simple at first, writing tests for it won't be as easy as we might think. There are some gotchas you have to avoid:

* The first (and the one you encounter) is you can't use `$this` as your tested instance in this test. The `$this`  represents the test being actually ran by atoum. You can't alter this one. You have to manually instanciate a test instance for your test case and work with it. Look at this line for an example : https://github.com/mikaelrandy/atoum/blob/feature/callStaticOnTestedClass/tests/units/classes/test.php#L174
* To be able to instanciate you own tested test, you will have to write a fake test class which you already did in your PR : https://github.com/mikaelrandy/atoum/blob/feature/callStaticOnTestedClass/tests/units/classes/test.php#L8
* Last but not least: creating a bug in the test class might fake your test suite result or avoid it from running correctly.